### PR TITLE
Adds streaming aggregation for OpenAI, fix metadata passing

### PR DESCRIPTION
**Pull request by jacoblee93**

### CODE REVIEW

**Comment by jacoblee93** [comment]
@hinthornw

**Comment by jacoblee93** [comment]
@agola11 need to test with tool calling

File: js/src/run_trees.ts [hunk 1]
```diff
@@ -156,7 +163,7 @@ export class RunTree implements BaseRun {
     const rt = new RunTree({
       name: props?.name ?? "<lambda>",
       parent_run: parentRun,
-      tags: deduppedTags,
+      tags: dedupedTags,
```
Comment (hinthornw) [inline js/src/run_trees.ts:166]: nice :)

File: js/src/wrappers/openai.ts [hunk 2]
```diff
@@ -60,6 +56,108 @@ type PatchedOpenAIClient<T extends OpenAIType> = {
...
+    }
+  }
+  const toolCalls: {
+    [
+      key: number
+    ]: Partial<OpenAI.Chat.Completions.ChatCompletionChunk.Choice.Delta.ToolCall>[];
+  } = {};
+  for (const c of choices) {
+    if (c.delta.content) {
+      message.content += c.delta.content;
```
Comment (hinthornw) [inline js/src/wrappers/openai.ts:80]: Discussed offline, but may want to check this works even if content were a list (not unlikely some time in the future)

**Comment by jacoblee93** [comment]
I think ready to go, final review?

File: js/src/wrappers/openai.ts [hunk 3]
```diff
@@ -60,6 +56,108 @@ type PatchedOpenAIClient<T extends OpenAIType> = {
...
+      for (const chunk of toolCallChunks) {
+        if (chunk.function) {
+          if (!message.tool_calls[idx].function) {
+            message.tool_calls[idx].function = {
+              name: "",
+              arguments: "",
+            };
+          }
+          if (chunk.function.name) {
+            message.tool_calls[idx].function.name += chunk.function.name;
```
Comment (hinthornw) [inline js/src/wrappers/openai.ts:120]: Shoudl this be overwrite instead of append?

**Review by hinthornw — APPROVED** [review]

**Pull request merged**